tests/test_state.py

```diff
@@ -196,4 +200,14 @@ def testContextSerialization(self):
         self.assertEqual( new_state.context['step'], 20)
         new_new_state = pickle.loads(new_new_file)
         self.assertEqual( new_new_state.context['step'], 30)
-        
+
+
+    def _test_state_gen_helper(self, name, msg):
+        self.assertEqual(name, 'statename')
+        self.assertEqual(msg, 'statemsg')
+        raise _CallbackExecuted
```

Comment:
The exception stops the processing in the event handling. For a single event this is not important. Though..

Consider setting a flag in the callback or assertMethodIsCalled (from https://stackoverflow.com/questions/3829742/assert-that-a-method-was-called-in-a-python-unit-test)